sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -396,6 +396,8 @@ case class Or(left: Expression, right: Expression) extends BinaryOperator with P
 
   override def sqlOperator: String = "OR"
 
+  override def nullable: Boolean = left.nullable && right.nullable
```

Comment:
Actually the current `def nullable: Boolean = left.nullable || right.nullable` for `Or` is not very accurate too.

At least when any of left/right is true, its `nullable` should be false.

Although we can't know if any of left/right is true under most of cases, but if they are foldable, then we can know that.

@gatorsmile @hvanhovell What do you think?